src/RegistryInterface.php

```diff
@@ -0,0 +1,25 @@
+<?php
+
+/**
+ * Part of the Joomla Framework Registry Package
+ *
+ * @copyright  Copyright (C) 2023 Open Source Matters, Inc.
+ * @license    GNU General Public License version 2 or later; see LICENSE
+ */
+
+namespace Joomla\Registry;
+
+/**
+ * Registry Interface
+ *
+ * @method exists(string $key)
```

Comment:
I really think these functions need to be written out and not just as comments. So I would deprecate the separator in the 2 branch version and remove it in version 3. So we can start using version 3 then in CMS 5.0. Better to have a clear cut somewhere than doing workarounds and introducing empty interfaces.

Reply:
Can be, but I thought it can go to 2.x.

Reply:
I am a strong advocate for preserving tradition while embracing progress. This means that I try to minimize breaking changes and avoid them whenever possible. However, in this instance, adding the shared public functions from both classes to the interface will not cause any disruptive changes. As a result, I believe it is appropriate to proceed with this update.

We need to be cautious about implementing return type declarations and function argument data types until we are fully prepared to transition to version 3, so the current Registry class must dictate the interface.

Regarding the deprecation of the separator in the 2 branch version, I'm not sure what you mean by this. @laoneo, could you clarify your intentions regarding the usage of the separator in the current Registry class? or are you speaking about the new flat class?

Reply:
Check the discussion above about the set function: I would love to see an interface like the following code (is not complete as I didn't add all the functions from the doc block:

```
interface RegistryInterface extends \JsonSerializable, \Countable
{
	public function exists($path);

	public function get($path, $default = null);

	public function set($path, $value);
}
```

Reply:
That's why the interface needs to go into 3.0 where we remove the separator argument from the registry class.

Reply:
The need to set the separator at the call level has been useful in many cases, specially with setting new data. Hmm maybe we are mixing 2 things that should not be mixed.